src/server/api/controllers/users.controller.js

```diff
@@ -8,14 +9,39 @@ const {
 
 const { isInteger } = require('../lib/utils/validators');
 
-const getUsers = () => {
-  return knex('users').select(
-    'users.id',
-    'users.fullName',
-    'users.email',
-    'users.createdOn',
-    'users.firebaseUId',
-  );
+const getUsers = async () => {
+  const result = await client.search({
+    index: usersIndex,
+    body: {
+      query: {
+        match_all: {},
+      },
+      size: 50,
+      sort: [
+        {
+          fullName: {
+            order: 'asc',
+          },
+        },
+      ],
+    },
+  });
+  /* eslint no-underscore-dangle: ["error", { "allow": ["_source", "_id"] }] */
+  return result.body.hits.hits.map((hit) => ({ ...hit._source, id: hit._id }));
+
+  // return knex('users').select(
+  //   'users.id',
+  //   'users.fullName',
+  //   'users.email',
+  //   'users.createdOn',
+  //   'users.firebaseUId',
+  // );
+  // KEEPING THIS SO THOSE NOT CONNECTED TO ELASTIC CAN USE getUsers IF NEEDED
```

Comment:
You could keep this code but in a way where you can conditionally check if the required ENV variables to set up Elasticsearch is present. If the required ENV variables are not set, you can then use this code to just look up the users directly from the DB.

Something else I'm noticing here is that this endpoint used to return id, fullName, email, createdOn, and firebaseUId; shouldn't the Elasticsearch variant also return the same thing? If so, it could make sense to also store the missing data in Elasticsearch.

Reply:
@orhantoy  struggling with this a bit. Attempted to use client.ping or client.info to get a boolean showing if it is OK, but can't seem to make it work to just return a clear status/boolean.
I think if I just check process.env - the variables might be there, but client still not connect? Would it be better to check the client in users.controller? if so, how?

Reply:
regarding the return, I considered it, but I suppose for the current version what we have is enough, as we are not using these other properties in any way at all.

Reply:
@orhantoy 
I am trying something like:
```javascript
const pingElastic = async () => {
  let elasticStatus;
  try {
    elasticStatus = await client.ping();
  } catch (err) {
    console.log(err);
  }

  console.log(`elastic ping: ${elasticStatus}`);
  return elasticStatus;
};

const elasticOK = pingElastic();

module.exports = { client, usersIndex, elasticOK };
```

and then 
```javascript
const getUsers = async () => {
  if (typeof elasticOK === 'undefined') {
    console.log('Elastic was not ok!');
    const DBusers = await knex('users').select(
      'users.id',
      'users.fullName',
      'users.email',
      'users.createdOn',
      'users.firebaseUId',
    );
    console.log(DBusers);
    return DBusers;
  } else {
    const result = await client.search({ ...
```
But if I test it with intentionally bad credentials, it does not get to the IF, it throws error:
              "reason": "missing authentication credentials for REST request [/users/_search]",
even though the elastic ping is logged as undefined.

Reply:
As we discussed, you can look more into having a mode where things still work if ES is not set up, in a follow-up PR.